### PR TITLE
Fetching upstream of 'master' from 'influxdata/telegraf'

### PR DESCRIPTION
**Pull request by JeffAshton**
D2L specific changes captured in:

https://github.com/Brightspace/telegraf/commit/c1edea14001c6f717426c0feaf9f0a670874c456

Updates `d2l_kinesis` to use `aws-sdk-go-v2`.

### CODE REVIEW

File: plugins/outputs/d2l_kinesis/d2lKinesisOutput.go [hunk 1]
```diff
@@ -21,6 +22,10 @@ const awsMaxRecordsPerRequest = 500
 const awsMaxRecordSize = 1048576  // 1 MiB
 const awsMaxRequestSize = 5242880 // 5 MiB
 
+type KinesisAPI interface {
```
Comment (JeffAshton) [inline plugins/outputs/d2l_kinesis/d2lKinesisOutput.go:25]: Use to come from `github.com/aws/aws-sdk-go/service/kinesis/kinesisiface` but they stopped doing this

File: plugins/outputs/d2l_kinesis/d2lKinesisOutput.go [hunk 2]
```diff
@@ -300,7 +309,10 @@ func (k *d2lKinesisOutput) putRecords(
 		return records
 	}
 
-	successfulRecordCount := int64(totalRecordCount) - *resp.FailedRecordCount
```
Comment (JeffAshton) [inline plugins/outputs/d2l_kinesis/d2lKinesisOutput.go:303]: `FailedRecordCount` use to be an `int64` but had a max value of 500

Old:

https://github.com/aws/aws-sdk-go/blob/961902f19e925368f0cd3059e1827817f9d9a05d/service/kinesis/api.go#L7470

New:

https://github.com/aws/aws-sdk-go-v2/blob/c07a63e84ec500ace7fefbf5391b9028f7615bd7/service/kinesis/api_op_PutRecords.go#L119

File: plugins/outputs/d2l_kinesis/d2lKinesisOutput.go [hunk 3]
```diff
@@ -275,7 +284,7 @@ func (k *d2lKinesisOutput) putRecords(
 
 	totalRecordCount := len(records)
 
-	entries := make([]*kinesis.PutRecordsRequestEntry, totalRecordCount)
```
Comment (JeffAshton) [inline plugins/outputs/d2l_kinesis/d2lKinesisOutput.go:278]: `PutRecordsRequestEntry` is really just 2 pointers and a `nil`, so they stopped doing the extra layer of pointer

https://github.com/aws/aws-sdk-go-v2/blob/c07a63e84ec500ace7fefbf5391b9028f7615bd7/service/kinesis/types/types.go#L152

**Review by daryl-mcmillan — APPROVED** [review]
🤷

**Comment by JeffAshton** [comment]
image

```
022-02-14T17:03:32Z D! [agent] Attempting connection to [outputs.d2l_kinesis]
2022-02-14T17:03:32Z D! [agent] Successfully connected to outputs.d2l_kinesis
2022-02-14T17:03:32Z D! [agent] Starting service inputs
2022-02-14T17:04:15Z D! [outputs.d2l_kinesis] Wrote 1 of 2 record(s) to Kinesis in 2.078222576s
2022-02-14T17:04:15Z D! [outputs.d2l_kinesis] Retrying 1 record(s)
2022-02-14T17:04:15Z D! [outputs.d2l_kinesis] Wrote 1 of 1 record(s) to Kinesis in 588.693719ms
2022-02-14T17:04:15Z D! [outputs.d2l_kinesis] Wrote batch of 100000 metrics in 3.809254935s
2022-02-14T17:04:15Z D! [outputs.d2l_kinesis] Buffer fullness: 8224 / 600000 metrics
2022-02-14T17:04:34Z D! [outputs.d2l_kinesis] Wrote 1 of 1 record(s) to Kinesis in 1.226357235s
2022-02-14T17:04:34Z D! [outputs.d2l_kinesis] Wrote batch of 51608 metrics in 1.81222399s
2022-02-14T17:04:34Z D! [outputs.d2l_kinesis] Buffer fullness: 5104 / 600000 metrics
2022-02-14T17:05:15Z D! [outputs.d2l_kinesis] Wrote 1 of 2 record(s) to Kinesis in 2.120973092s
2022-02-14T17:05:15Z D! [outputs.d2l_kinesis] Retrying 1 record(s)
2022-02-14T17:05:15Z D! [outputs.d2l_kinesis] Wrote 1 of 1 record(s) to Kinesis in 608.457928ms
2022-02-14T17:05:15Z D! [outputs.d2l_kinesis] Wrote batch of 100000 metrics in 3.925699406s
2022-02-14T17:05:15Z D! [outputs.d2l_kinesis] Buffer fullness: 9736 / 600000 metrics
2022-02-14T17:05:34Z D! [outputs.d2l_kinesis] Wrote 1 of 1 record(s) to Kinesis in 1.228567325s
2022-02-14T17:05:34Z D! [outputs.d2l_kinesis] Wrote batch of 53120 metrics in 1.842118427s
2022-02-14T17:05:34Z D! [outputs.d2l_kinesis] Buffer fullness: 5104 / 600000 metrics
```

**Pull request merged**